File: 백준/Silver/11723. 집합/집합.py
import sys

input = lambda: sys.stdin.readline().rstrip()

if __name__ == '__main__':
    S = 0
    for _ in range(int(input())):
        q, *data = input().split()
        if q == 'add':
            x = int(data[0])
            S |= 1 << x
        elif q == 'remove':
            x = int(data[0])
            S &= ~(1 << x)
        elif q == 'check':
            x = int(data[0])
            print(int(S & (1 << x) != 0))
        elif q == 'toggle':
            x = int(data[0])
            if S & (1 << x):
                S &= ~ (1 << x)
            else:
                S |= (1 << x)
        elif q == 'all':
            S = (1 << 21) - 1
        elif q == 'empty':
            S = 0

File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>

using namespace std;

int main()
{
    ios::sync_with_stdio(NULL);
    cin.tie(NULL);
    cout.tie(NULL);

    int M;
    cin >> M;
    int s = 0;

    while (M--)
    {
        string command;
        cin >> command;

        if (command == "add")
        {
            int num;
            cin >> num;
            s |= 1 << num;
        }
        else if (command == "remove")
        {
            int num;
            cin >> num;
            s &= ~(1 << num);
        }
        else if (command == "check")
        {
            int num;
            cin >> num;
            if (s & (1 << num))
                cout << "1\n";
            else
                cout << "0\n";
        }
        else if (command == "toggle")
        {
            int num;
            cin >> num;
            if (s & (1 << num))
                s &= ~(1 << num);
            else
                s |= (1 << num);
        }
        else if (command == "all")
        {
            s = (1 << 21) - 1;
        }
        else if (command == "empty")
        {
            s = 0;
        }
    }
}